airbyte-workers/src/main/java/io/airbyte/workers/temporal/SyncWorkflow.java

```diff
@@ -141,21 +151,62 @@ public StandardSyncOutput replicate(JobRunConfig jobRunConfig,
         return syncInput;
       };
 
-      final TemporalAttemptExecution<StandardSyncInput, StandardSyncOutput> temporalAttemptExecution = new TemporalAttemptExecution<>(
-          workspaceRoot,
-          jobRunConfig,
-          getWorkerFactory(sourceLauncherConfig, destinationLauncherConfig, jobRunConfig, syncInput),
-          inputSupplier,
-          new CancellationHandler.TemporalCancellationHandler());
+      final Predicate<ReplicationOutput> shouldAttemptAgain =
+          output -> output.getReplicationAttemptSummary().getStatus() != ReplicationStatus.COMPLETED;
 
-      return temporalAttemptExecution.get();
+      final BiFunction<StandardSyncInput, ReplicationOutput, StandardSyncInput> nextAttemptInput = (input, lastOutput) -> {
+        final StandardSyncInput newInput = Jsons.clone(input);
+        newInput.setState(lastOutput.getState());
+        return newInput;
+      };
+
+      final PartialSuccessTemporalAttemptExecution<StandardSyncInput, ReplicationOutput> temporalAttemptExecution =
+          new PartialSuccessTemporalAttemptExecution<>(
+              workspaceRoot,
+              jobRunConfig,
+              getWorkerFactory(sourceLauncherConfig, destinationLauncherConfig, jobRunConfig, syncInput),
+              inputSupplier,
+              new CancellationHandler.TemporalCancellationHandler(),
+              shouldAttemptAgain,
+              nextAttemptInput,
+              MAX_RETRIES);
+
+      // aggregate each attempts output into a sync summary.
+      // todo (cgardens) - this operation is lossy (we lose the ability to see the amount of data
+      // replicated by each attempt). likely in the future, we will want to retain this info and surface
+      // it.
+      final List<ReplicationOutput> attemptOutputs = temporalAttemptExecution.get();
```

Comment:
can we put this in a separate method for readability?

Reply:
reduceReplicationOutputs